src/__experimental__/components/input/input-presentation.style.js

```diff
@@ -0,0 +1,68 @@
+import styled, { css } from 'styled-components';
+import PropTypes from 'prop-types';
+import baseTheme from '../../../style/themes/base';
+import OptionsHelper from '../../../utils/helpers/options-helper';
+import sizes from './input-sizes.style';
+import inputClassicStyling from './input-presentation-classic.style';
+
+const InputPresentationStyle = styled.div`
+  align-items: center;
+  background: transparent;
+  border: 1px solid ${({ theme }) => theme.colors.border};
+  box-sizing: border-box;
+  cursor: text;
+  display: flex;
+  flex-wrap: wrap;
+  flex: 1;
+  margin: 0px;
+  min-height: ${({ size }) => sizes[size].height};
+  padding-left: ${({ size }) => sizes[size].padding};
+  padding-right: ${({ size }) => sizes[size].padding};
+  width: 100%;
+
+  ${({ disabled, theme }) => disabled && css`
+    background: ${theme.disabled.input};
+    border-color: ${theme.disabled.border};
+    cursor: not-allowed;
+  `}
+  ${({ readOnly }) => readOnly && css`
+    background: transparent !important;
+    border-color: transparent !important;
+  `}
+  ${({ hasFocus, theme }) => hasFocus && css`
+    && { outline: 3px solid ${theme.colors.focus}; }
+  `}
+  ${stylingForValidation('info')}
+  ${stylingForValidation('warning')}
+  ${stylingForValidation('error')}
+
+  ${inputClassicStyling}
```

Comment:
Will this end up printing a whole bunch of component-specific css to the browser (namely the rules generated above this line), that is then overridden? Do we care about that?

Reply:
that will happen only when a theme is set to `classic`

Reply:
With classic theme enabled, it will print both, with the latter overriding. It's a minor trade-off I suppose for it being a temporary theme.

Reply:
Fully aware that it'll only happen when theme is set to classic. Given that most of our components in the near future will be using classic theme, does it seem as minor when you consider that there is likely to be a majority of components rendering duplicate `css`, rather than just all of the inputs rendered on a page? 

Something like:
```
${(props) => {
  if(props.theme.name === 'classic') return inputClassicStyling(props):
  return css`/* non classic css */`;
}
```

Reply:
The problem here is that the overrides are only over _some_ of the attributes, as few changes as possible, while still relying on the original styling. In most cases this will be the same (some padding, colours, or sizing slightly different). The DLS is not that far different from what classic is today. In some of the components I have done a "if classic else this" (such as here https://github.com/Sage/carbon/blob/f986289744f8ea2678821746e296529f29814c83/src/__experimental__/components/form-field/form-field.style.js#L9 ), but for InputPresentation in particular it seemed like an easier trade off to drop it in a single block. 

It may be a case of using both approaches, whichever makes more sense given the context?